src/App.js

```diff
@@ -209,6 +239,7 @@ const App = ({ locationId }) => {
                   navigate("/");
                 } else {
                   navigate(`/locations/${location.id}`);
+                  animateToCoordinates(location);
```

Comment:
It would be nice if we could avoid having to separately navigate+animate and instead have the animation happen "automatically". I think we could do that by putting the `animateToCoordinates` call inside a `useEffect` hook that is dependent on the `selectedLocation` state. Then I think you also wouldn't need `isInitialLocationCentred`.

I also find it a bit odd that we aren't already setting the viewport based on `selectedLocation`. Does that mean that theoretically if you were linked to a location that is outside our default map view we wouldn't pan to it? I guess that doesn't matter in practice now because all the locations are in the default viewport?

Reply:
Regarding your second point, correct, I never implemented that logic, and suggested that @foreverhomes should implement it here. I only just recently added routing to link to locations, and missed this as something to add.

Reply:
Although this raises an interesting point... do we still need that animate call after navigating to the new location? Could we get rid of the `isInitialLocationCentered` flag and just always animate when we navigate to a new selected location?